src/entities/bounds.rs

```diff
@@ -0,0 +1,37 @@
+pub struct BoundingBox {
+    pub width: i32,
+    pub height: i32,
+    pub x: i32,
+    pub y: i32
+}
+
+impl BoundingBox {
+    pub fn new(x: i32, y: i32, width: i32, height: i32) -> BoundingBox {
+        BoundingBox{x: x, y: y, width: width, height: height}
+    }
+    pub fn update_position(&mut self, x: i32, y: i32) {
+        self.x = x;
+        self.y = y;
+    }
+    pub fn collides_with(&self, candidate: &BoundingBox) -> bool { 
+        ! (self.top() > candidate.bottom() ||
```

Comment:
Just remembered why it wouldn't work with AND logic. We need to check all values to determine a collision. Essentially, all conditions must equate to false to be considered a collision. Obviously, if one rectangle is on top of the other this logic fails but for our purposes we're not worried about that case.